### PR TITLE
R doc 697 final

### PR DESCRIPTION
**Pull request by maximburyak**
That's the result of the long awaited subscription documentation.
I'll be out of office for the next 1.5 weeks, so take in consideration that any fix requests (I guess there would be some) could be treated with delay

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/what-are-data-subscriptions.dotnet.markdown [hunk 1]
```diff
@@ -1,30 +1,115 @@
...
+* Data subscription provides a reliable and handy way to perform documents processing on the client side.  
 
-When you run a subscription worker it will send you all documents matching the specified creation options, it may be a raw query, or lambdas describing the filtering and transformation. Processing progress will be persisted and therefore it can be paused and resumed from the last point it was stopped. Documents are sent in batches and progress will be registered only after the whole batch is processed and acknowledged. Documents are always sent in Etag order which means that data that already been processed and acknowledged won't be sent twice. Nevertheless, there are conditions in which documents could be sent twice: 
+* Documents that match a pre-defined criteria are sent in batches from the server to the client. 
+  The documents' data can be transformed before being send. 
+  
+* Note: If the database has Revisions defined, then special handling can be done on the server before sending the data to the client.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/what-are-data-subscriptions.dotnet.markdown]: what handling?

File: Documentation/4.0/Raven.Documentation.Pages/glossary/subscription-creation-options.markdown [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+| **Query** | string | The query |
+| **QueryParameters** | Parameters (Dictionary&lt;string, object&gt;) | The query parameters |
+| **Start** | int | The start of records to read |
+| **PageSize** |  int | The page size |
+| **WaitForNonStaleResults** | bool | If set to true, the server side will wait until the results are non-stale or until a timeout. |
+| **WaitForNonStaleResultsTimeout** | TimeSpan? | The timeout for WaitForNonStaleResults |
+| **CutoffEtag** | long? | The cutoff Etag is used to check if the index has already processed a document with the given Etag. |
+| **DisableCaching** | bool | Indicates if the query results should be read from the cache (if cached previously), or added to the cache (if there were no cached items prior). |
+| **SkipDuplicateChecking** | bool | Allows to skip duplicate checking during queries. |
+| **ExplainScores** | bool | When a query result should contain an explanation about how docs are scored against a query. |
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/glossary/subscription-creation-options.markdown]: I removed it as we already have it defined in API Overview. BTW we'll probably drop the glossary section

File: Raven.Documentation.Web/Raven.Documentation.Web.csproj [hunk 3]
```diff
@@ -13,7 +13,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>Raven.Documentation.Web</RootNamespace>
     <AssemblyName>Raven.Documentation.Web</AssemblyName>
-    <TargetFrameworkVersion>v4.7.1</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
```
Comment (arekpalinski) [inline Raven.Documentation.Web/Raven.Documentation.Web.csproj]: I reverted this change. Please install .net 4.7.1

File: Raven.Documentation.Web/Web.config [hunk 4]
```diff
@@ -1,64 +1,72 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (arekpalinski) [inline Raven.Documentation.Web/Web.config]: I reverted those changes

File: Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown [hunk 5]
```diff
@@ -0,0 +1,54 @@
+﻿# Revisions support
+---
+
+{NOTE: }
+
+Data subscription supports subscribing not only on documents, but also on all their revisions.  
+Revision support should be defined in the subscription, it also requires revisions to be configured on the colleciton in question.
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown]: `colleciton` -> `collection`

File: Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+In this page:  
+[Revisions processing](#revisions-processing-order)  
+[Simple declaration and usage](#simple-declaration-and-usage)   
+[Revisions processing and projection](#revisions-processing-and-projection)  
+
+{NOTE/}
+
+{PANEL:Revisions processing order}
+Documents revisions will be processed in paris, meaning if a document was changed 6 times in a row, subscription will process 6 times 6 pairs of versions of that document.
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown]: `paris` -> `pairs`

Comment (ayende) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown]: And I find this statement confusing.

File: Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+{WARNING For the subscription revisions to work properly, it's crucial to make sure that the revisioning configuration stores documents revisions enough time, without discarding unprocessed revisions/}
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown]: Need to explain why this is the case

File: Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/MaintainanceOperations.markdown [hunk 8]
```diff
@@ -0,0 +1,103 @@
+﻿# Maintainance operation
+---
+
+{NOTE: }
+This page covers data subscriptions maintainance operations:  
+[Deletion](#deletion)
```
[the remark this answers is not in the document]
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/MaintainanceOperations.markdown]: ok

File: Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/what-are-data-subscriptions.dotnet.markdown (hunk 1 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/what-are-data-subscriptions.dotnet.markdown]: ok

File: Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown (hunk 6 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown]: ok

File: Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown (hunk 7 above)
Comment (maximburyak) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/data-subscriptions/AdvancedTopics/SubscriptionWithRevisioning.markdown]: ok

**Pull request merged**